### PR TITLE
Josef.melkus/BAAS-270 boost remove and package

### PR DESCRIPTION
**Pull request by Melky-Phoe**
Added install and packages

### CODE REVIEW

**Comment by Nothrax** [comment]
Your install does not work, what I did: I installed smurf into local folder with those commands:

```
cmake .. -DBRINGAUTO_TESTS=ON -DCMAKE_INSTALL_PREFIX=./install -DBRINGAUTO_PACKAGE=ON -DCMLIB_DIR=/home/nothrax/Documents/bringauto/cmakelib -DCMAKE_BUILD_TYPE=Release
make -j 8
make install
```

I created example project and added library in CMakeLists.txt with this:

``` 
SET(state_smurf_DIR /home/nothrax/Documents/bringauto/bakalarky/StateSmurf/_build/install/lib/cmake/state_smurf)
FIND_PACKAGE(state_smurf )
TARGET_LINK_LIBRARIES(smurfTest PUBLIC diagram_smurf_lib transition_smurf_lib)
```

And I got message:
error

also when I use ldd on libtransition_smurf_lib.so I get message:
Screenshot from 2022-03-25 18-02-37

File: SmurfExampleApp/CMakeLists.txt [hunk 1]
```diff
@@ -19,22 +18,23 @@ FIND_PACKAGE(libbringauto_logger)
...
 			--scenario ${CMAKE_SOURCE_DIR}/test/smurf/scenarios.json
-			--executable ${CMAKE_BINARY_DIR}/smurfExampleApp
-			--evaluator ${STATESMURF_DIR}/SmurfEvaluator/_build/smurfEvaluator
+			--executable ${CMAKE_BINARY_DIR}/smurf_example_app
+			--evaluator ${STATESMURF_DIR}/SmurfEvaluator/_build/smurf_evaluator
 			--output ${CMAKE_SOURCE_DIR}/test/smurf/
 			#--create-etalons
-			)
+			)]]
+	# TODO put this in a script and remove from here
```
Comment (Nothrax) [inline SmurfExampleApp/CMakeLists.txt:39]: todo left

File: SmurfExampleApp/CMakeLists.txt [hunk 2]
```diff
@@ -19,22 +18,23 @@ FIND_PACKAGE(libbringauto_logger)
...
 
-ADD_EXECUTABLE(smurfExampleApp main.cpp)
-TARGET_LINK_LIBRARIES(smurfExampleApp PUBLIC SmurfExampleAppLib)
+ADD_EXECUTABLE(smurf_example_app main.cpp)
+TARGET_LINK_LIBRARIES(smurf_example_app PUBLIC smurf_example_app_lib)
 
-IF(BRINGAUTO_TESTS AND STATESMURF_TESTS)
-	ENABLE_TESTING()
+IF(BRINGAUTO_TESTS)
+	#[[ENABLE_TESTING()
```
Comment (Nothrax) [inline SmurfExampleApp/CMakeLists.txt:29]: commented code

File: SmurfExampleApp/CMakeLists.txt [hunk 3]
```diff
@@ -19,22 +18,23 @@ FIND_PACKAGE(libbringauto_logger)
...
-	SET(STATESMURF_DIR ${CMAKE_SOURCE_DIR}/..)
+	SET(STATESMURF_DIR ${CMAKE_CURRENT_LIST_DIR}/..)
 	ADD_TEST(NAME automatedEvaluatorTests COMMAND ${Python3_EXECUTABLE} ${STATESMURF_DIR}/SmurfTestingScripts/CompareScenarios.py
 			--scenario ${CMAKE_SOURCE_DIR}/test/smurf/scenarios.json
-			--executable ${CMAKE_BINARY_DIR}/smurfExampleApp
-			--evaluator ${STATESMURF_DIR}/SmurfEvaluator/_build/smurfEvaluator
+			--executable ${CMAKE_BINARY_DIR}/smurf_example_app
+			--evaluator ${STATESMURF_DIR}/SmurfEvaluator/_build/smurf_evaluator
 			--output ${CMAKE_SOURCE_DIR}/test/smurf/
 			#--create-etalons
```
Comment (Nothrax) [inline SmurfExampleApp/CMakeLists.txt:37]: commented line

**Comment by Melky-Phoe** [comment]
bringauto_logger was set as individual requirement, state_smurf doesn't install it.

File: CMakeLists.txt [hunk 4]
```diff
@@ -1,9 +1,79 @@
...
+OPTION(BRINGAUTO_SYSTEM_DEP "Enable system dependencies" OFF)
+
+IF (BRINGAUTO_PACKAGE)
+    IF(NOT BRINGAUTO_INSTALL)
+        SET(BRINGAUTO_INSTALL ON CACHE BOOL "Forced install due to BRINGAUTO_PACKAGE=ON" FORCE)
+        MESSAGE(WARNING "BRINGAUTO_INSTALL is switched to on because of BRINGAUTO_PACKAGE=ON")
+    ENDIF()
+ENDIF()
+
+#SET(CMAKE_SKIP_BUILD_RPATH OFF)
```
Comment (Nothrax) [inline CMakeLists.txt:22]: commented code

**Comment by Nothrax** [comment]
When creating package, there is still problem with linking, ba loger is not found and the diagram smurf has really strange path
Screenshot from 2022-03-28 17-56-08

**Review by Nothrax — APPROVED** [review]

**Pull request merged**